samplesheets/tests/test_views_taskflow.py

```diff
@@ -1829,6 +1837,385 @@ def test_accept_lock_failure(self):
+
+        with self.login(self.user):
+            response = self.client.post(
+                reverse(
+                    'samplesheets:irods_request_accept',
+                    kwargs={'irodsdatarequest': obj.sodar_uuid},
+                ),
+                {'confirm': True},
+            )
+            self.assertRedirects(
+                response,
+                reverse(
+                    'samplesheets:irods_requests',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+            )
+            self.assertEqual(
+                list(get_messages(response.wsgi_request))[-1].message,
+                'iRODS data request "{}" accepted.'.format(
+                    obj.get_display_name()
+                ),
+            )
+
+        obj.refresh_from_db()
+        self.assertEqual(obj.status, 'ACCEPTED')
+        self._assert_alert_count(CREATE_ALERT, self.user, 0)
+        self._assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        self._assert_alert_count(ACCEPT_ALERT, self.user, 0)
+        self._assert_alert_count(ACCEPT_ALERT, self.user_delegate, 0)
+        self.assert_irods_obj(coll_path, False)
+        self.assert_irods_obj(obj_path, False)
+
+
+class TestIrodsRequestAcceptBatchView(TestIrodsRequestViewsBase):
+    """Tests for IrodsRequestAcceptBatchView"""
+
+    def test_render(self):
+        """Test rendering IrodsRequestAcceptBatchView"""
+        self.assert_irods_obj(self.path)
+        self.assert_irods_obj(self.path_md5)
+
+        with self.login(self.user_contrib):
+            self.client.post(
+                reverse(
+                    'samplesheets:irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+            self.assertEqual(IrodsDataRequest.objects.count(), 1)
+            self.client.post(
+                reverse(
+                    'samplesheets:irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data2,
+            )
+            self.assertEqual(IrodsDataRequest.objects.count(), 2)
+
+        with self.login(self.user):
+            response = self.client.post(
+                reverse(
+                    'samplesheets:irods_request_accept_batch',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                {
+                    'irods_requests': ','.join(
+                        [
+                            str(irods_request.sodar_uuid)
+                            for irods_request in IrodsDataRequest.objects.all()
+                        ]
+                    ),
+                },
+            )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.context['affected_object_paths'],
+            sorted(
+                set(
```

Comment:
Is `set()` actually needed here? Are there multiple requests with the same path?

Reply:
When I created multiple requests refering to the same path directly from shell, there was possibility of accepting multiple requests with only one path (so it should be filtered with set). I thought, that different users might want to delete the same object and so would create redundant requests, which in that case should be filtered. 

If it's not the case I can delete set() :)

Reply:
Ah, good point. You can leave the implementation as is.